### PR TITLE
guard against isdir EACESS in completions

### PR DESCRIPTION
**Pull request by IanButterworth**
Fixes https://github.com/JuliaLang/julia/issues/54154

### CODE REVIEW

File: ext/REPLExt/completions.jl [hunk 1]
```diff
@@ -37,7 +37,13 @@ function complete_expanded_local_dir(s, i1, i2, expanded_user, oldi2)
     cmp = REPL.REPLCompletions.complete_path(s, i2, shell_escape=true)
     cmp2 = cmp[2]
     completions = [REPL.REPLCompletions.completion_text(p) for p in cmp[1]]
-    completions = filter!(x -> isdir(s[1:prevind(s, first(cmp2)-i1+1)]*x), completions)
+    completions = filter!(completions) do x
+        try
+            isdir(s[1:prevind(s, first(cmp2)-i1+1)]*x)
```
Comment (KristofferC) [inline ext/REPLExt/completions.jl:42]: Shouldn't this function return `true` or `false`? Does it always do that now or is the fall through returning `nothing` which would error in `filter!`?

Comment (IanButterworth) [reply ext/REPLExt/completions.jl:42]: It throws if it doesn't have perms to stat the path

Comment (KristofferC) [reply ext/REPLExt/completions.jl:42]: Yes, and it should return `false` in that case, but right now it isn't I think?

Comment (IanButterworth) [reply ext/REPLExt/completions.jl:42]: @vtjnash perhaps you have an opinion on what to do here?

**Review by vtjnash — COMMENTED** [review]
Yes, the catch needs to end with a bool so that the filter doesn't error

**Comment by IanButterworth** [comment]
Oh right, I thought @KristofferC was talking about changing the behavior of `isdir` not the filter anonymous function.

File: ext/REPLExt/completions.jl [hunk 2]
```diff
@@ -37,7 +37,13 @@ function complete_expanded_local_dir(s, i1, i2, expanded_user, oldi2)
...
+        catch e
+            (e isa Base.IOError && e.code == Base.UV_EACCES) || rethrow()
```
Comment (IanButterworth) [inline ext/REPLExt/completions.jl:44]:
```suggestion
            (e isa Base.IOError && e.code == Base.UV_EACCES) || rethrow()

            return false
```

**Review by vtjnash — COMMENTED** [review]
Though really I think we should later add a kwarg to these functions because this pattern is getting really annoying

Eg `isdir(path, default=false)`
Or `default=nothing` or `default=throw` (the current behavior) or `default=Returns`, or such

File: ext/REPLExt/completions.jl [hunk 3]
```diff
@@ -37,7 +37,14 @@ function complete_expanded_local_dir(s, i1, i2, expanded_user, oldi2)
...
+            return false
```
Comment (IanButterworth) [inline ext/REPLExt/completions.jl:45]:
```suggestion
            if e isa Base.IOError && e.code == Base.UV_EACCES
                return false
            else
                rethrow()
            end
```

**Pull request merged**